src/slices/users.slice.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+
+type Users = {
+  users: User[] | null;
+};
+
+const initialState: Users = {
+  users: null,
+};
+
+export const fetchAllUsers = createAsyncThunk<User[], void>(
+  'users/fetchAllUsers',
+  async () => {
+    try {
+      const result = await getUsers();
+
+      return result;
+    } catch (error) {
+      throw error;
```

Comment:
The catch block is rethrowing the error without any additional handling or logging. Consider adding some error handling logic or logging to provide more context or to handle specific error cases.